AutomatedTesting/Gem/PythonTests/assetpipeline/ap_fixtures/debug_parse_and_compare.py

```diff
@@ -0,0 +1,207 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import os
+import re
+from typing import List
+import ly_test_tools.o3de.pipeline_utils as utils
+import warnings
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def populate_expected_product_assets(assets_to_validate: List[str]):
+    # Check that each given source asset resulted in the expected jobs and products.
+    expected_product_list = []
+    for expected_source in assets_to_validate:
+        for expected_job in expected_source.jobs:
+            for expected_product in expected_job.products:
+                expected_product_list.append(expected_product.product_name)
+
+    return expected_product_list
+
+
+def trim_floating_point_values_from_same_length_lists(diff_actual: List[str], diff_expected: List[str],
```

Comment:
Are you going to update fbx_tests.py to use these as part of this pull request later, or will that be a separate pull request?